graphql_compiler/cost_estimation/cardinality_estimator.py

```diff
@@ -69,17 +69,103 @@ def _get_last_edge_direction_and_name_to_location(location):
     return edge_direction, edge_name
 
 
-def _get_parent_and_child_name_from_edge(schema_graph, child_location):
+def _get_base_classnames_of_parent_and_child_from_edge(schema_graph, child_location):
     """Get the base classname of a location and its parent from the last edge information."""
     edge_direction, edge_name = _get_last_edge_direction_and_name_to_location(child_location)
     edge_element = schema_graph.get_edge_schema_element_or_raise(edge_name)
     if edge_direction == INBOUND_EDGE_DIRECTION:
-        parent_name_from_edge = edge_element.base_out_connection
-        child_name_from_edge = edge_element.base_in_connection
+        parent_base_classname = edge_element.base_out_connection
+        child_base_classname = edge_element.base_in_connection
     elif edge_direction == OUTBOUND_EDGE_DIRECTION:
-        parent_name_from_edge = edge_element.base_in_connection
-        child_name_from_edge = edge_element.base_out_connection
-    return parent_name_from_edge, child_name_from_edge
+        parent_base_classname = edge_element.base_in_connection
+        child_base_classname = edge_element.base_out_connection
+    else:
+        raise AssertionError(u'Expected edge direction to be either inbound or outbound.'
+                             u'Found: edge {} with direction {}'.format(edge_name, edge_direction))
+    return parent_base_classname, child_base_classname
+
+
+def _get_outbound_and_inbound_names_from_locations(query_metadata, child_location, parent_location):
```

Comment:
nit: Any particular reason for the `child_location, parent_location` order of arguments? So long as we're consistent across all functions that take those args, I guess it's fine, though that would not have been the order I would have assumed a priori.

Reply:
Merely due to consistency. 
I'd be happy to change this in another PR once this gets merged. How does that sound?

Reply:
Sure. Or you could add something about "refactoring" to the title and save yourself a round-trip through the CI bot :) All the same to me.